src_bagl/ui_flow_signMessage712.c

```diff
@@ -1,62 +1,87 @@
-#include "shared_context.h"
-#include "ui_callbacks.h"
+#ifdef HAVE_EIP712_FULL_SUPPORT
 
-void prepare_domain_hash_v0() {
-    snprintf(strings.tmp.tmp, 70, "0x%.*H", 32, tmpCtx.messageSigningContext712.domainHash);
-}
+#include "ui_logic.h"
+#include "shared_context.h"  // strings
+
+enum { UI_712_POS_REVIEW, UI_712_POS_END };
+static uint8_t ui_pos;
 
-void prepare_message_hash_v0() {
-    snprintf(strings.tmp.tmp, 70, "0x%.*H", 32, tmpCtx.messageSigningContext712.messageHash);
+static void dummy_cb(void) {
```

Comment:
Comment to explain why we need a dummy CB and what it actually does

Reply:
It is used with the _dummy_ UX_STEP which came from the need to have some logic run between the _dynamic_ step (which displays the actual content of the message) and the _approve_ & _reject_ steps.
* If the entire message hasn't been received and there is more to come, going right on the _dynamic_ step should trigger the next data to come and be processed, and then come back to the _dynamic_ step once it has been loaded with the new data.
* If we already reached the end of the message, the step should be a pass-through and let the user go to the _approve_ step